lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (giovannibattistasciortino@gmail.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage RHSM repositories using the C(subscription-manager) command
+description:
+    - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+version_added: "2.4"
+author: "Giovanni Sciortino (giovannibattistasciortino@gmail.com)"
+notes:
+    - In order to manage rhsm repositories the system must be already registered to rhsm manually or using the ansible module redhat_subscription.
+requirements:
+    - subscription-manager
+options:
```

Comment:
I have chosen to use the current options (enable, disable, list and disable_first) to allow users to enable a set of repositories and disable all the others using a single tasks (I think this is a common need and a valid single subscription-manager command). If a use a single task the result variable changed will report correctly the value False(also in check mode) when I will execute the task on a host where this task has been already applied.

Instead using the Ansible-like options, I have difficult to disable and enable repositories using a single task but I have to split them in two tasks as in the following example and they will return always the result changed=True when applied multiple times on the same host.

```
- redhat_repository: name=* state=disabled 
- redhat_repository: name=rhel-6-server-rpms state= enabled
```
What do you think about this? Is it a good reason to use the current options or do you prefer Ansible-like parameters?